packages/ai-ide-agents/src/browser/content-replacer.ts

```diff
@@ -17,20 +17,22 @@
 export interface Replacement {
     oldContent: string;
     newContent: string;
+    multiple?: boolean;
 }
 
 export class ContentReplacer {
     /**
      * Applies a list of replacements to the original content using a multi-step matching strategy.
      * @param originalContent The original file content.
      * @param replacements Array of Replacement objects.
+     * @param allowMultiple If true, all occurrences of each oldContent will be replaced. If false, an error is returned when multiple occurrences are found.
      * @returns An object containing the updated content and any error messages.
      */
     applyReplacements(originalContent: string, replacements: Replacement[]): { updatedContent: string, errors: string[] } {
         let updatedContent = originalContent;
         const errorMessages: string[] = [];
 
-        replacements.forEach(({ oldContent, newContent }) => {
+        replacements.forEach(({ oldContent, newContent, multiple }) => {
```

Comment:
Do we want to guard the case where the same oldContent appears multiple times in the suggestions with different newContent?

Reply:
Yes, good point